dissect/target/tools/query.py

```diff
@@ -110,67 +118,127 @@ def main():
                 "The --rewrite-cache option will be ignored as --no-cache or --only-read-cache are specified",
             )
 
+    # -l/-f follow the same code path, a list = a list of functions + a display command
+    # this makes sure the behavior will always be in sync
+    if args.list is not False:
+        args.function = args.list
+
+    # If no functions (or list) then we cannot continue at all
+    if not args.function:
+        parser.error("argument -f/--function is required")
+
+    # No targets is allowed, but only for lists/help
+    if not args.targets and args.list is False and "-h" not in rest and "--help" not in rest:
+        parser.error("too few arguments")
+
+    # In that case the user wants to see all available plugins and loaders
+    # and we turn off compatibility checks, because they will always fail
+    # we don't raise an exception though, because the user might want to 'amend'
+    # the command later...
+    if not args.targets:
+        targets = [Target()]
+        args.compatible = False
+    else:
+        targets = Target.open_all(args.targets, args.children)
+
+    # To check whether the output is uniform and which plugins we need to load
+    # either for the list or for execution, we collect all the work to be done here
+    tasklist = []
+    for target in targets:
+        if args.child:
+            try:
+                target = target.open_child(args.child)
+            except Exception:
+                continue
+        functions = target.find_plugin_functions(args.function, args.compatible)
+        tasklist.append((target, functions))
```

Comment:
Not a big fan of this approach. If you execute anything in a directory with 1000+ targets in it, it'll first attempt to load all of those targets before doing anything.

Reply:
We probably only want this wwhen `--compatible` is specified, so it is explicit that this may be an expensive operation.

Reply:
Is there no way to avoid this approach completely? I'd much prefer that.